orm/index.go

```diff
@@ -400,3 +468,179 @@ func (i Index) insert(db weave.KVStore, index []byte, pk []byte) error {
 
 	return db.Set(key, save)
 }
+
+const nativeIdxPrefix = "_x."
+
+func NewNativeIndex(name string, indexer MultiKeyIndexer) Index {
+	return &nativeIndex{
+		name:    name,
+		indexer: indexer,
+	}
+}
+
+type nativeIndex struct {
```

Comment:
Please add a small comment here to prevent the need of reading the code to understand what this thing is.